[ BUG ] Unit tests fail to run

## Describe the bug

Attempting to execute unit tests results in the following error:

```python3
    @property
    def token_status(self) -> int:
        """Provide the current token_status."""
>       return self.auth_object.token_status
E       AttributeError: 'CustomIOA' object has no attribute 'auth_object'

../../../Library/Caches/pypoetry/virtualenvs/caracara-ksX5R94U-py3.11/lib/python3.11/site-packages/falconpy/_service_class/_base_service_class.py:193: AttributeError
============================================================================================================= short test summary info ==============================================================================================================
FAILED tests/unit_tests/test_hosts.py::test_describe_devices - AttributeError: 'CustomIOA' object has no attribute 'auth_object'
FAILED tests/unit_tests/test_hosts.py::test_describe_devices__online_only - AttributeError: 'CustomIOA' object has no attribute 'auth_object'
[...]
```

This is because FalconPy checks that the `auth_object` given to a Service Class is an instance of `FalconInterface`
```python3
if auth_object:
    if isinstance(auth_object, FalconInterface):  # Issue 1043
```
and the current usage of `MagicMock` doesn't pass this check. The following patch fixes the problem
```diff
+++ b/tests/unit_tests/test_custom_ioas.py
@@ -25,7 +25,7 @@ def client():
     """Pytest fixture that provides a client with a mocked OAuth2 object, so no API credentials
     are needed.
     """
-    auth = MagicMock(autospec=falconpy.OAuth2)
+    auth = MagicMock(spec=falconpy.OAuth2, autospec=True)
     client = Client(falconpy_authobject=auth)
     return client
```
as the following snippet demonstrates
```python3
>>> auth = MagicMock(autospec=falconpy.OAuth2)
>>> isinstance(auth, falconpy.FalconInterface)
False
>>> auth = MagicMock(spec=falconpy.OAuth2, autospec=True)
>>> isinstance(auth, falconpy.FalconInterface)
True
```

## To Reproduce

`pytest tests/unit_tests`

## Expected behavior

The unit tests execute successfully

## Environment

### Operating System Version

macOS Sonoma 14.6.1

### Python Version

Python 3.11.9

### Poetry Version

Poetry (version 1.8.4)

### Python Package Versions
<details>
<summary>Package versions collapsed, click to see</summary>

```
caracara-filters==1.0.0
certifi==2024.8.30
charset-normalizer==3.4.0
click==8.1.7
coverage==7.6.1
crowdstrike-falconpy==1.4.5
dill==0.3.9
flake8==7.1.1
idna==3.10
inflate64==1.0.0
iniconfig==2.0.0
isort==5.13.2
markdown-it-py==3.0.0
mccabe==0.7.0
mdurl==0.1.2
multivolumefile==0.2.3
mypy-extensions==1.0.0
packaging==24.1
pathspec==0.12.1
pbr==6.1.0
platformdirs==4.3.6
pluggy==1.5.0
prompt_toolkit==3.0.48
psutil==6.0.0
py7zr==0.22.0
pybcj==1.0.2
pycodestyle==2.12.1
pycryptodomex==3.21.0
pydocstyle==6.3.0
pyflakes==3.2.0
Pygments==2.18.0
pylint==3.2.7
pyppmd==1.1.0
pytest==8.3.3
python-dateutil==2.9.0.post0
PyYAML==6.0.2
pyzstd==0.16.2
requests==2.32.3
rich==13.9.2
six==1.16.0
snowballstemmer==2.2.0
stevedore==5.3.0
tabulate==0.9.0
texttable==1.7.0
toml==0.10.2
tomlkit==0.13.2
urllib3==2.2.3
wcwidth==0.2.13
```
</details>

## Additional context

Somewhat related: https://github.com/CrowdStrike/falconpy/issues/1043#issuecomment-1715731268

--- a/tests/unit_tests/test_custom_ioas.py
+++ b/tests/unit_tests/test_custom_ioas.py
@@ -25,7 +25,7 @@ def client():
     """Pytest fixture that provides a client with a mocked OAuth2 object, so no API credentials
     are needed.
     """
-    auth = MagicMock(autospec=falconpy.OAuth2)
+    auth = MagicMock(spec=falconpy.OAuth2, autospec=True)
     client = Client(falconpy_authobject=auth)
     return client
 

--- a/tests/unit_tests/test_hosts.py
+++ b/tests/unit_tests/test_hosts.py
@@ -115,7 +115,7 @@ def hosts_test():
     def decorator(func):
         @patch("caracara.modules.hosts.hosts.HostGroup", autospec=falconpy.HostGroup)
         @patch("caracara.modules.hosts.hosts.Hosts", autospec=falconpy.Hosts)
-        @patch("caracara.client.OAuth2")
+        @patch("caracara.client.OAuth2", autospec=True)
         def new_func(mock_oauth2, mock_hosts, mock_hostgroup):
             # B106 is a bandit warning for hardcoded passwords.
             # This is a testing context and the credentials passed to this constructor

--- a/tests/unit_tests/test_prevention_policies.py
+++ b/tests/unit_tests/test_prevention_policies.py
@@ -17,7 +17,7 @@ def prevpol_test():
             "caracara.modules.prevention_policies.prevention_policies.PreventionPolicies",
             autospec=falconpy.PreventionPolicies,
         )
-        @patch("caracara.client.OAuth2")
+        @patch("caracara.client.OAuth2", autospec=True)
         def test_new_func(mock_oauth2, mock_prevpol):
             # B106 is a bandit warning for hardcoded passwords.
             # This is a testing context and the credentials passed to this constructor

--- a/tests/unit_tests/test_response_policies.py
+++ b/tests/unit_tests/test_response_policies.py
@@ -17,7 +17,7 @@ def respol_test():
             "caracara.modules.response_policies.response_policies.ResponsePolicies",
             autospec=falconpy.ResponsePolicies,
         )
-        @patch("caracara.client.OAuth2")
+        @patch("caracara.client.OAuth2", autospec=True)
         def test_new_func(mock_oauth2, mock_prevpol):
             # B106 is a bandit warning for hardcoded passwords, this is a testing context and
             # the credentials passed to this constructor aren't valid and aren't used.

